### PR TITLE
fix(backend): fixes healthz response by adding json content type. Fixes #7525

### LINKED ISSUE #7525: [backend] health service response should be consumed by application/json not text/plain
### Environment

*  How did you deploy Kubeflow Pipelines (KFP)? install via kubeflow manifest with version v1.4
*  KFP version: v1.7.0
*  KFP SDK version: v1.7.0 (golang version under backend/api/go_http_client)

### Steps to reproduce

1. start pipeline backend api server by listening port 8888
2. sent http request with swagger generated code stub with the following code example

```

package main

import (
	"github.com/go-openapi/strfmt"
	"github.com/go-openapi/runtime"
	healthz_client "github.com/kubeflow/pipelines/backend/api/go_http_client/healthz_client"
	healthz_service "github.com/kubeflow/pipelines/backend/api/go_http_client/healthz_client/healthz_service"
)

func main() {
	healthServiceClient := healthz_client.New(
		client.New(
			"localhost:8888",
			healthz_client.DefaultBasePath,
			[]string{"http"}),
		strfmt.NewFormats(),
	)
	
	clientAuthInfoWriter := runtime.ClientAuthInfoWriterFunc(func(clientRequest runtime.ClientRequest, registry strfmt.Registry) error {
		return nil
	})
	
	resp, err := p.HealthServiceClient.HealthzService.GetHealthz(healthz_service.NewGetHealthzParams, clientAuthInfoWriter)
        if err != nil {
            // should not be here
        }
}


```

3. When I was sent healthz endpoint (`/apis/v1beta1/healthz`) with go_http_client(swagger generated code stubs), the generated code stub failed to decode the response with the error message `&{false} (*healthz_model.APIGetHealthzResponse) is not supported by the TextConsumer, can be resolved by supporting TextUnmarshaler interface`

the http log are displayed below:

```
GET /apis/v1beta1/healthz HTTP/1.1
Host: 127.0.0.1:8888
User-Agent: Go-http-client/1.1
Accept: application/json
Accept-Encoding: gzip


HTTP/1.1 200 OK
Content-Length: 96
Content-Type: text/plain; charset=utf-8
Date: Wed, 06 Apr 2022 06:41:40 GMT

{"commit_sha":"6ccf5bcd0b9db955c91caab85fa130714527f414", "tag_name":"1.7.0", "multi_user":true}
```

however, the message was wrapped into a json struct but the content type was specified with `text/plain`. 
This triggered [APIGetHealthzResponse](https://github.com/kubeflow/pipelines/blob/master/backend/api/go_http_client/healthz_model/api_get_healthz_response.go#L16) failed to decode the response, as it failed to implement `TextUnmarshaler` interface.

The server code (on v1.7.0 as well as on master) also confirmed this https://github.com/kubeflow/pipelines/blob/74c7773ca40decfd0d4ed40dc93a6af591bbc190/backend/src/apiserver/main.go#L146, without proper set the content-type, the default one will determined by [http.DetectContentType](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/sniff.go;drc=refs%2Ftags%2Fgo1.18;l=21) which will be 'text/plain'

### Expected result

the golang example mentioned earier with swagger generated stub should work well.

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

###
if this was confirmed as bug, I would like to submit a patch for this. Thanks

### PR DESCRIPTION
**Pull request by hsinhoyeh**
**Description of your changes:**
1. explicit content type with application/json on healthz response.
2. add healthz client api under common/client for integration test
3. add integration test for healthz endpoint

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by hsinhoyeh** [comment]
few thoughts for this PR:
1. I don't think we should touch [WaitForReady] (https://github.com/kubeflow/pipelines/blob/master/backend/test/test_utils.go#L39) during this PR as its primary job is to ensure the backend api is ready for test. It is better to pass the raw http.Response for better assessing its status.
2. According to previous point, so I added a test suite for Healthz, hope it will not over-tested.

File: backend/test/integration/healthz_api_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+func (s *HealthzApiTest) cleanUp() {
+}
+
+func (s *HealthzApiTest) TestHealthzAPI() {
+	t := s.T()
+
+	/* ---------- Verify healthz response ---------- */
+	healthzResp, err := s.healthzClient.GetHealthz()
+	assert.Nil(t, err)
+	assert.NotNil(t, healthzResp)
```
Comment (zijianjoy) [inline backend/test/integration/healthz_api_test.go:59]: Can we check the content within healthzResp are json parsable?

Comment (zijianjoy) [reply backend/test/integration/healthz_api_test.go:59]: To show more about what I mean: For example: you can check the content within healthzResp contains `multi_user` key and value: https://github.com/kubeflow/pipelines/blob/74c7773ca40decfd0d4ed40dc93a6af591bbc190/backend/api/python_http_client/docs/ApiGetHealthzResponse.md.

Comment (hsinhoyeh) [reply backend/test/integration/healthz_api_test.go:59]: @zijianjoy thanks for the reply, but I think I still don't get it.
the question here is `healthzResp` is a well-defined structure, you can check the structure definition here 
https://github.com/kubeflow/pipelines/blob/74c7773ca40decfd0d4ed40dc93a6af591bbc190/backend/api/go_client/healthz.pb.go#L44
so the field `multi-user` will be there. 

But I think the question is: are we able to verify the field value `multi-user` to be true in a multi-tenancy setup. (as the multi-user field is a simple boolean field, we are unable to verify when the value is not presented.)
And still, it is hard to verify during this test (as we should be able to create a test environment w/o multiuser).

**Comment by zijianjoy** [comment]
/lgtm
/approve 

Thank you for your change! @hsinhoyeh

**Pull request merged**